fontforge/python.c

```diff
@@ -1739,16 +1750,65 @@ return( reto );
 /* ************************************************************************** */
 /* Points */
 /* ************************************************************************** */
-static PyObject *PyFFPoint_dup(PyFF_Point *self) {
-    PyFF_Point *ret = (PyFF_Point *)PyFF_PointType.tp_alloc(&PyFF_PointType, 0);
-    if ( ret==NULL )
+
+static PyFF_Point *PyFFPoint_CNew(double x, double y, int on_curve, int sel, char *name) {
+    /* Convenience routine for creating a new point from C */
+    PyFF_Point *self = (PyFF_Point *)PyFF_PointType.tp_alloc(&PyFF_PointType, 0);
+    if ( self==NULL )
 	return( NULL );
-    ret->x = self->x;
-    ret->y = self->y;
-    ret->on_curve = self->on_curve;
-    ret->selected = self->selected;
-    ret->name = copy(self->name);
-return( (PyObject *) ret );
+    self->x = x;
+    self->y = y;
+    self->on_curve = on_curve;
+    self->selected = sel;
+    self->name = copy(name);
+    return( self );
+}
+
+static PyObject *PyFFPoint_dup(PyFF_Point *self) {
+    PyFF_Point *ret = PyFFPoint_CNew(self->x, self->y, self->on_curve,
+                                     self->selected, self->name);
+    return( (PyObject *) ret );
+}
+
+static PyFF_Point *PyFFPoint_Parse(PyObject *args, bool dup, bool always) {
+    double x,y;
+    PyFF_Point *p=NULL;
+    int i, on, sel;
+
+    if ( args==NULL && !always )
+	return NULL;
+
+    x = y = 0.0;
+    on = true;
+    sel = false;
+    if ( !PyArg_ParseTuple( args, "(ddii)", &x, &y, &on, &sel )) {
```

Comment:
Some of these embedded-tuple combos are probably pointless but they were supported by insertPoint()